bundle/python/transform.go

```diff
@@ -0,0 +1,106 @@
+package python
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/artifacts"
+	"github.com/databricks/cli/bundle/libraries"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+)
+
+// This mutator takes the wheel task and trasnforms it into notebook
+// which installs uploaded wheels using %pip and then calling corresponding
+// entry point.
+func TransformWheelTask() bundle.Mutator {
+	return &transform{}
+}
+
+type transform struct {
+}
+
+func (m *transform) Name() string {
+	return "python.TransformWheelTask"
+}
+
+const INSTALL_WHEEL_CODE = `%%pip install --force-reinstall %s`
+
+const NOTEBOOK_CODE = `
+%%python
+%s
+
+from contextlib import redirect_stdout
+import io
+import sys
+sys.argv = [%s]
+
+import pkg_resources
+_func = pkg_resources.load_entry_point("%s", "console_scripts", "%s")
+
+f = io.StringIO()
+with redirect_stdout(f):
+	_func()
+s = f.getvalue()
+dbutils.notebook.exit(s)
+`
```

Comment:
I can imagine us doing the same thing for a file task or a notebook task. For both, the prelude where the libraries are installed will be the same. Can we reuse the same code for those, and then have a different trampoline for each of them? E.g. for wheel tasks, we do the exec as done here. For file tasks, we do an eval (or equivalent). For notebook tasks, we prepend the cells to the notebook and use that as result.

Reply:
@kartikgupta-db Backdrop for the discussion we had yesterday about setting the $PYTHONPATH, working directory, and more. If we can model these as generic middleware and not tie it to wheels/notebooks/etc specifically then we can reuse across the board.

Reply:
@pietern @kartikgupta-db now we can reuse parameterised `mutator.NewTrampoline` function which handles all internal logic needed to generate trampoline notebook.

Python trampoline for wheel can be a reference implementation